tests/framework/clients/object.go

```diff
@@ -34,11 +40,46 @@ func CreateObjectOperation(rookRestClient contracts.RestAPIOperator) *ObjectOper
 //ObjectCreate Function to create a object store in rook
 //Input paramatres -None
 //Output - output returned by rook Rest API client
-func (ro *ObjectOperation) ObjectCreate(storeName string, replicaCount int32) (string, error) {
+func (ro *ObjectOperation) ObjectCreate(namespace, storeName string, replicaCount int32, callAPI bool, k8sh *utils.K8sHelper) error {
 	store := model.ObjectStore{Name: storeName, Gateway: model.Gateway{Instances: replicaCount, Port: model.RGWPort}}
 	store.DataConfig.ReplicatedConfig.Size = 1
 	store.MetadataConfig.ReplicatedConfig.Size = 1
-	return ro.restClient.CreateObjectStore(store)
+	if callAPI {
+		logger.Infof("creating the object store via the rest API")
+		_, err := ro.restClient.CreateObjectStore(store)
+		return err
+	}
+
+	logger.Infof("creating the object store via CRD")
+	storeSpec := fmt.Sprintf(`apiVersion: rook.io/v1alpha1
+kind: ObjectStore
+metadata:
+  name: %s
+  namespace: %s
+spec:
+  metadataPool:
+    replicated:
+      size: 1
```

Comment:
lets use replica Count parameter  instead of hardcoding the size to 1

Reply:
The replicaCount actually should be the number of rgw pod instances, so the name is a bit confusing here.